File: java_kotlin/src/main/kotlin/283_move_zeroes.kt
// https://leetcode.com/problems/move-zeroes/
fun moveZeroes(nums: IntArray): Unit {
    val size = nums.size
    if (size < 2) return
    var index = 0
    var nonZeroIndex = nums.indexOfFirst { it != 0 }
    if (nonZeroIndex == -1) return
    while (nonZeroIndex != size) {
        val temp = nums[index]
        nums[index++] = nums[nonZeroIndex]
        nums[nonZeroIndex] = temp
        if (index == size) return
        while (++nonZeroIndex != size && nums[nonZeroIndex] == 0) {}
    }
}

File: go/solutions/283_move_zeroes.go
package solutions

// https://leetcode.com/problems/move-zeroes/
func moveZeroes(nums []int)  {
    newIndex := 0
	for i := 0; i < len(nums); i++ {
        if nums[i] != 0 {
            nums[newIndex] = nums[i]
            newIndex = newIndex + 1
        }
	}
    for ; newIndex < len(nums); newIndex++ {
        nums[newIndex] = 0
    }
}
